tests/python/gaia-ui-tests/gaiatest/apps/messages/regions/report.py

```diff
@@ -0,0 +1,44 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from marionette_driver import expected, By, Wait
+from gaiatest.apps.base import Base
+from gaiatest.form_controls.header import GaiaHeader
+
+class Report(Base):
+
+    _text_type = (By.CSS_SELECTOR, 'p[data-l10n-id="message-type-sms"]')
+    _time_sent_locator = (By.CSS_SELECTOR, 'span.sent-timestamp')
+    _time_received_locator = (By.CSS_SELECTOR, 'span.received-timestamp')
+    _header_locator = (By.ID, 'information-report-header')
+
+    _sender_locator = (By.CSS_SELECTOR, 'span[data-l10n-id="report-from-title"] ~ .contact-list .js-contact-info')
+    _receiver_locator = (By.CSS_SELECTOR, 'span[data-l10n-id="report-to-title"] ~ .contact-list .js-contact-info')
```

Comment:
Yeah, this locator is complex, but we're sure that we get either the receiver of the sender.

This locator takes every contacts in the list, I believe. The fact that we use `find_element` instead of `find_elements` makes it retrieving the first one.